### PR TITLE
Fix DirectX (and some OpenXR) bindings & regenerate

### LINKED ISSUE #521: Silk.NET.DXGI.AdapterDesc returning wrong values
# Summary
I'm trying to get the `DedicatedSystemMemory` and `DedicatedVideoMemory` values from the current Adapter but all the values in the `AdapterDesc` object are very wrong:

2021 06 11-18 08 04  rider64_aLjPQrMhB0

Here are the actuall values it should return (using `SharpDX.DXGI`):
2021 06 11-18 27 55  rider64_n3FEjGwZVl

# Steps to reproduce
- Platform: Windows 10
- Framework Version: .NET 5.0
- API: DXGI

Used code:

```csharp
unsafe
{
    using var api = DXGI.GetApi();
    
    IDXGIFactory1* factory1 = default;
    
    try
    {
        SilkMarshal.ThrowHResult(api.CreateDXGIFactory1(SilkMarshal.GuidPtrOf<IDXGIFactory1>(), (void**)&factory1));
        
        uint i = 0u;
        while (true)
        {
            IDXGIAdapter1* adapter1 = default;
            var res = factory1->EnumAdapters1(i, &adapter1);
            
            var exception = Marshal.GetExceptionForHR(res);
            if (exception != null) break;

            AdapterDesc1 adapterDesc = default;
            SilkMarshal.ThrowHResult(adapter1->GetDesc1(&adapterDesc));
            
            var systemMemory = (ulong) adapterDesc.DedicatedSystemMemory;
            var videoMemory = (ulong) adapterDesc.DedicatedVideoMemory;

            adapter1->Release();
            i++;
        }
    }
    catch (Exception e)
    {
        Utils.ErrorThrow(e);
    }
    finally
    {
        
        if (factory1->LpVtbl != (void**)IntPtr.Zero)
            factory1->Release();
    }
}
```

# Comments
Thread in `#suppport` on the Discord starts here: [Link](https://discord.com/channels/521092042781229087/607634593201520651/852941141232844850)

### LINKED ISSUE #523: Enum-count array parameters not parsed correctly

### PR DESCRIPTION
**Pull request by Perksey**
Closes #521 
Closes #523

### CODE REVIEW

File: src/Core/Silk.NET.Core/Native/Structs/Luid.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    public struct Luid
+    {
+        public uint Low;
+        public int High;
+
+        // TODO REMOVE THESE IN 3.0! THESE ARE FOR COMPATIBILITY TO SOFTEN THE BLOW OF THE BREAKING CHANGE IN 2.6!
+        public static implicit operator (ulong, long)(Luid instance) => (instance.Low, instance.High);
+
+        public static implicit operator Luid((ulong, long) instance)
+            => new() {Low = (uint) instance.Item1, High = (int) instance.Item2};
```
Comment (HurricanKai) [inline src/Core/Silk.NET.Core/Native/Structs/Luid.cs:15]: :/ I mean anyone using these tuples had broken code anyways?

File: src/OpenXR/Silk.NET.OpenXR/Structs/UuidMSFT.gen.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+{
+    [NativeName("Name", "XrUuidMSFT")]
+    public unsafe partial struct UuidMSFT
+    {
+        /// <summary></summary>
+        [NativeName("Type", "uint8_t")]
+        [NativeName("Type.Name", "uint8_t")]
+        [NativeName("Name", "bytes")]
+        public fixed byte Bytes[16];
+    }
```
Comment (HurricanKai) [inline src/OpenXR/Silk.NET.OpenXR/Structs/UuidMSFT.gen.cs:27]: = GUID? or is this different?

File: src/OpenXR/Silk.NET.OpenXR/Structs/SceneSphereBoundMSFT.gen.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+using Silk.NET.Core.Attributes;
+using Silk.NET.Core.Contexts;
+using Silk.NET.Core.Loader;
+
+#pragma warning disable 1591
+
+namespace Silk.NET.OpenXR
+{
+    [NativeName("Name", "XrSceneSphereBoundMSFT")]
+    public unsafe partial struct SceneSphereBoundMSFT
```
Comment (HurricanKai) [inline src/OpenXR/Silk.NET.OpenXR/Structs/SceneSphereBoundMSFT.gen.cs:20]: Sphere<float> has the same layout iirc.

File: src/OpenXR/Silk.NET.OpenXR/Structs/ScenePlanesMSFT.gen.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+{
+    [NativeName("Name", "XrScenePlanesMSFT")]
+    public unsafe partial struct ScenePlanesMSFT
+    {
+        public ScenePlanesMSFT
+        (
+            StructureType? type = StructureType.TypeScenePlanesMsft,
+            void* next = null,
+            uint? scenePlaneCount = null,
+            ScenePlaneMSFT* scenePlanes = null
```
Comment (HurricanKai) [inline src/OpenXR/Silk.NET.OpenXR/Structs/ScenePlanesMSFT.gen.cs:27]: For reference on board: Struct overloads! (@Perksey you can ignore 😅)

File: src/OpenXR/Extensions/Silk.NET.OpenXR.Extensions.OCULUS/OculusAudioDeviceGuid.gen.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+namespace Silk.NET.OpenXR.Extensions.OCULUS
+{
+    [Extension("XR_OCULUS_audio_device_guid")]
+    public unsafe partial class OculusAudioDeviceGuid : NativeExtension<XR>
+    {
+        public const string ExtensionName = "XR_OCULUS_audio_device_guid";
+        /// <summary>To be documented.</summary>
+        [NativeApi(EntryPoint = "xrGetAudioInputDeviceGuidOculus")]
+        public partial Result GetAudioInputDeviceGuidOcul([Count(Count = 0)] Instance instance, [Count(Count = 0)] wchar_t buffer);
```
Comment (HurricanKai) [inline src/OpenXR/Extensions/Silk.NET.OpenXR.Extensions.OCULUS/OculusAudioDeviceGuid.gen.cs:25]: wchar_t?

File: src/OpenXR/Silk.NET.OpenXR/Structs/SceneMeshVertexBufferMSFT.gen.cs [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+/// <summary></summary>
+        [NativeName("Type", "uint32_t")]
+        [NativeName("Type.Name", "uint32_t")]
+        [NativeName("Name", "vertexCountOutput")]
+        public uint VertexCountOutput;
+/// <summary></summary>
+        [NativeName("Type", "XrVector3f*")]
+        [NativeName("Type.Name", "XrVector3f")]
+        [NativeName("Name", "vertices")]
+        public Vector3f* Vertices;
```
Comment (HurricanKai) [inline src/OpenXR/Silk.NET.OpenXR/Structs/SceneMeshVertexBufferMSFT.gen.cs:81]: Well I guess I missed this in some earlier review, very unfortunate, Vector3f should be Vector3 instead.

**Review by HurricanKai — CHANGES_REQUESTED** [review]

File: src/OpenXR/Silk.NET.OpenXR/Structs/SceneMeshVertexBufferMSFT.gen.cs (hunk 6 above)
Comment (Perksey) [reply src/OpenXR/Silk.NET.OpenXR/Structs/SceneMeshVertexBufferMSFT.gen.cs:81]: Yep and I opened an issue for this recently

File: src/OpenXR/Extensions/Silk.NET.OpenXR.Extensions.OCULUS/OculusAudioDeviceGuid.gen.cs (hunk 5 above)
Comment (Perksey) [reply src/OpenXR/Extensions/Silk.NET.OpenXR.Extensions.OCULUS/OculusAudioDeviceGuid.gen.cs:25]: Good catch, seems we haven’t mapped that yet

File: src/OpenXR/Silk.NET.OpenXR/Structs/SceneSphereBoundMSFT.gen.cs (hunk 3 above)
Comment (Perksey) [reply src/OpenXR/Silk.NET.OpenXR/Structs/SceneSphereBoundMSFT.gen.cs:20]: Will fix.

File: src/OpenXR/Silk.NET.OpenXR/Structs/UuidMSFT.gen.cs (hunk 2 above)
Comment (Perksey) [reply src/OpenXR/Silk.NET.OpenXR/Structs/UuidMSFT.gen.cs:27]: Should just be GUID so yeah will fix.

File: src/Core/Silk.NET.Core/Native/Structs/Luid.cs (hunk 1 above)
Comment (Perksey) [reply src/Core/Silk.NET.Core/Native/Structs/Luid.cs:15]: True, will remove.

**Comment by Perksey** [comment]
Ready for re-review.

**Review by HurricanKai — APPROVED** [review]

**Pull request merged**